src/segment-loader.js

```diff
@@ -1948,6 +1948,20 @@ export default class SegmentLoader extends videojs.EventTarget {
 
     segmentInfo.waitingOnAppends = 0;
 
+    // segments with no data
+    if (!segmentInfo.hasAppendedData_) {
+      // override settings for metadata only segments
+      segmentInfo.timestampOffset = segmentInfo.startOfSegment;
+      segmentInfo.timingInfo = {start: 0};
+      segmentInfo.waitingOnAppends++;
+
+      // update the timestampoffset
+      this.updateSourceBufferTimestampOffset_(segmentInfo);
```

Comment:
Since there's no `handleData_` call (there being no data), should we call to `processMetadataQueue_ ` here for segments that have no video/audio data but have id3 tags/captions?

Reply:
I think that we should yeah.